hack/bats/tests/preserve-env.bats

```diff
@@ -135,16 +125,93 @@ local_setup() {
     assert_line FOO=foo
     assert_line FOOBAR=foobar
     assert_line BAR=bar
-    refute_line --regexp '^BARBAZ='
+    assert_line BARBAZ=barbaz
 }
 
-@test 'setting both allow list and block list generates a warning' {
-    export LIMA_SHELLENV_ALLOW=FOO
-    export LIMA_SHELLENV_BLOCK=BAR
+@test 'wildcard patterns work in all positions and combinations' {
+    # Test wildcard at middle, and other combinations
+    export LIMA_SHELLENV_BLOCK="FOO*BAR,*FOO*BAR*,*TEST*,*SUFFIX"
     export FOO=foo
-    run -0 --separate-stderr limactl shell --preserve-env "$NAME" printenv FOO
-    assert_output foo
-    assert_stderr --regexp 'level=warning msg="Both LIMA_SHELLENV_BLOCK and LIMA_SHELLENV_ALLOW are set'
+    export FOOBAR=foobar
+    export FOOXYZBAR=fooxyzbar
+    export FOOBAZ=foobaz
+    export BAZBAR=bazbar
+    export BAR=bar
+    export XFOOYBARZDOTCOM=xfooybarzdotcom
+    export PREFIX_TEST_VAR=prefix_test_var
+    export VAR_SUFFIX=var_suffix
+    export NORMAL_VAR=normal_var
+    export UNRELATED=unrelated
+    run -0 limactl shell --preserve-env "$NAME" printenv
+    
+    # Should block FOO*BAR pattern
+    refute_line --regexp '^FOOBAR='
+    refute_line --regexp '^FOOXYZBAR='
+    
+    # Should block *FOO*BAR* pattern
+    refute_line --regexp '^XFOOYBARZDOTCOM='
+    
+    # Should block *TEST* and *SUFFIX patterns
+    refute_line --regexp '^PREFIX_TEST_VAR='
+    refute_line --regexp '^VAR_SUFFIX='
+    
+    # Should allow variables that don't match any pattern
+    assert_line FOO=foo
+    assert_line FOOBAZ=foobaz
+    assert_line BAZBAR=bazbar
+    assert_line BAR=bar
+    assert_line NORMAL_VAR=normal_var
+    assert_line UNRELATED=unrelated
+}
+
+@test 'comprehensive allow/block interaction with wildcards and default blocklist' {
+    # Test allowlist with wildcards, and other test rules
+    export LIMA_SHELLENV_ALLOW="SSH_FOO,CUSTOM*,FOO*,*PREFIX,MIDDLE*PATTERN,SUFFIX*"
```

Comment:
Again, I think this is too much. Keep the number of test cases as low as possible while making sure all distinct situations are covered.

When I wrote previously that there are too many tests, I didn't mean to cram them together into a single `@test`, but rather to carefully curate them to cover each edge case, but remove all redundancy.